packages/monaco-editor-wrapper/src/editorClassic.ts

```diff
@@ -15,41 +15,71 @@ export type MonacoLanguageExtensionConfig = {
     mimetypes?: string[];
 }
 
-export type EditorClassicConfig = {
+export type EditorAppConfigClassic = EditorAppConfig & {
+    editorAppType: 'classic';
+    automaticLayout?: boolean;
+    theme?: string;
+    editorOptions?: editor.IStandaloneEditorConstructionOptions;
+    diffEditorOptions?: editor.IStandaloneDiffEditorConstructionOptions;
     languageExtensionConfig?: MonacoLanguageExtensionConfig;
     languageDef?: languages.IMonarchLanguage;
     themeData?: editor.IStandaloneThemeData;
-}
+};
 
-export class EditorClassic extends MonacoEditorBase implements MonacoEditorWrapper {
+export class EditorAppClassic extends EditorAppBase {
 
-    async init() {
-        const wrapperConfig = this.monacoConfig as EditorClassicConfig;
+    constructor(id: string, userConfig: UserConfig) {
+        super(id, userConfig);
+        const userInput = userConfig.wrapperConfig.editorAppConfig as EditorAppConfigClassic;
+        // default to vs-light
+        this.getAppConfig().theme = userInput.theme ?? 'vs-light';
+        // default to true
+        this.getAppConfig().automaticLayout = userInput.automaticLayout ?? true;
+
+        this.editorOptions = userInput.editorOptions ?? {};
+        this.editorOptions.automaticLayout = userInput.automaticLayout ?? true;
+
+        this.diffEditorOptions = userInput.diffEditorOptions ?? {};
+        this.diffEditorOptions.automaticLayout = userInput.automaticLayout ?? true;
+
+        this.getAppConfig().languageExtensionConfig = userInput.languageExtensionConfig ?? undefined;
+        this.getAppConfig().languageDef = userInput.languageDef ?? undefined;
+        this.getAppConfig().themeData = userInput.themeData ?? undefined;
+    }
 
+    getAppType(): EditorAppType {
+        return 'classic';
+    }
+
+    getAppConfig(): EditorAppConfigClassic {
+        return this.appConfig as EditorAppConfigClassic;
+    }
+
+    async init() {
         // register own language first
-        const extLang = wrapperConfig?.languageExtensionConfig;
+        const extLang = this.getAppConfig().languageExtensionConfig;
```

Comment:
Some of these usages here are `this.getAppConfig()` while others are `this.appConfig`. If we need the appConfig it would be good to grab it once and use it for the duration of the function, or change the usages to be consistent (all one or the other).

Reply:
The base class only uses the abstract method `getConfig` that is implemented by both app classes.